include/seastar/parquet/exception.hh

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <exception>
+#include <sstream>
+
+namespace parquet {
+
+class parquet_exception : public std::exception {
+    std::string _msg;
+public:
+    ~parquet_exception() throw() override {}
+
+    static parquet_exception eof(const std::string& msg = "") {
+        std::stringstream ss;
+        ss << "Unexpected end of stream";
+        if (!msg.empty()) {
+            ss << ": " << msg << ".";
+        }
+        return parquet_exception(ss.str());
```

Comment:
nitpick - it's an exception, so it's considered to be rare, so it's fine to use a helper format function, which makes the code more readable:
```cpp
return parquet_exception(format("Unexpected end of stream: {}", msg));
```
Also, let's not use default arguments - that applies to all exceptions below as well. Using a default empty msg makes debugging hard, and lack of a default argument will force engineers to put something meaningful in the message.

Reply:
Changed in e23d38a.